rebasehelper/output_tool.py

```diff
@@ -19,216 +19,177 @@
 #
 # Authors: Petr Hracek <phracek@redhat.com>
 #          Tomas Hozza <thozza@redhat.com>
+from __future__ import print_function
 
-import sys
 import os
 import six
-import json
+import pkg_resources
 
-from rebasehelper.exceptions import RebaseHelperError
-from rebasehelper.logger import LoggerHelper, logger, logger_report
+from rebasehelper.logger import logger
 from rebasehelper.results_store import results_store
-
-output_tools = {}
-
-
-def register_output_tool(output_tool):
-    output_tools[output_tool.PRINT] = output_tool
-    return output_tool
+from colors import red, green, yellow
 
 
 class BaseOutputTool(object):
-
     """
-    Class used for testing and other future stuff, ...
-    Each method should overwrite method like run_check
+    Base class for OutputTools.
+    print_cli_summary must be overridden in order to produce different CLI output
     """
 
     DEFAULT = False
+    NAME = 'name'
 
     @classmethod
-    def match(cls, cmd=None, *args, **kwargs):
-        """Checks if tool name matches the desired one."""
-        raise NotImplementedError()
+    def get_report_path(cls, app):
+        return os.path.join(app.results_dir, 'report.' + cls.NAME)
 
-    def print_summary(self, path, results, **kwargs):
+    @classmethod
+    def print_cli_summary(cls, app):
         """
-        Return list of files which has been changed against old version
-        This will be used by checkers
+        Print report of the rebase
+
+        :param app: Application instance
         """
-        raise NotImplementedError()
+        cls.app = app
+        cls.print_patches_cli()
 
+        cls.colored_output(yellow, '\nAvailable logs:')
+        print('{0}:\n{1}'.format('Debug log', app.debug_log_file))
+        if results_store.get_old_build() is not None:
+            print('{0}:\n{1}'.format('Old build logs and (S)RPMs', os.path.join(app.results_dir, 'old-build')))
+        if results_store.get_new_build() is not None:
+            print('{0}:\n{1}'.format('New build logs and (S)RPMs', os.path.join(app.results_dir, 'new-build')))
+        print()
 
-@register_output_tool
-class TextOutputTool(BaseOutputTool):
+        cls.colored_output(yellow, '%s:' % 'Rebased sources')
+        print("%s" % app.rebased_sources_dir)
 
-    """ Text output tool. """
+        cls.colored_output(yellow, '%s:' % 'Patch containing changes')
```

Comment:
Like in the report file, wouldn't something like `Patch with differences` sound a bit better? Or maybe just `Generated patch`.